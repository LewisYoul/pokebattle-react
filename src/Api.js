import React, { Component } from 'react';


export class Api extends Component {
  constructor() {
    super();
    this.state = {
      players: [],
    };
  }

componentDidMount(){
  fetch("http://localhost:4000/api/players/players.json")
  .then(results => {
    return results.json();
  }).then(data => {
    let players = data.map((player) => {
      return(
        <div>
          <h2>{player.name}</h2>
          <h4>{player.hp}</h4>
          <h4>{player.action}</h4>
        </div>
      )
    })
    this.setState({players: players});
  })
}

  render(){
    return (
      <div>{this.state.players}</div>
    )
  }


}

export default Api;
