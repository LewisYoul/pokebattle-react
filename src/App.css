@import url('https://fonts.googleapis.com/css?family=Bangers|Russo+One');

*{
  margin: 0 auto;
  padding: 0;
  box-sizing: inherit;
}


body {
  box-sizing: border-box;
  padding-bottom: 2cm;
  background-color: #d63232;
}

.player{
  border-radius: 20px;
}

.versus {
  font-family: Bangers;
  font-size: 35px;
  text-align: center;
  width: 620px;
   height: 100px;
   position: absolute;
   top: 70%;
   left: 50%;
   margin-left: -310px;
   margin-top: -50px;
   color: #19717e;
   -webkit-text-stroke-width: 0.02px;
   -webkit-text-stroke-color: #fdea34;
   text-shadow: 2px 5px rgb(32, 30, 31);
}

.fight {
  font-family: Bangers;
  font-size: 35px;
  color: yellow;
  position: absolute;
  left: 31%;
  top: 100%;
  width: 200px;
  padding: 10px 20px;
  margin-bottom: 1cm;
  outline: none;
  border: 2px solid #fdea34;
  background-color: #19717e;
  box-shadow: 6px 8px #1f2226;

}

.fight:active{
  transform: translateY(0.3rem);
  box-shadow: 0.3rem 0.3rem #1f2226;
}


.reset{
  font-family: Bangers;
  font-size: 35px;
  color: yellow;
  position: absolute;
  left: 51%;
  top: 100%;
  width: 200px;
  padding: 10px 20px;
  outline: none;
  border: 2px solid #fdea34;

  background-color: #19717e;
  box-shadow: -5px 8px #1f2226;
}

.reset:active{
  transform: translateY(0.3rem);
  box-shadow: -0.2rem 0.3rem #1f2226;
}

.move{
  position: absolute;
  left: 30%;
}

.a{
  position: absolute;
  top: 28%;
  left: 36%;

}

.character{
  display: inline-block;
  position: absolute;
  top: 68%;
  left: 8%;
}

.title{
  position: absolute;
  top: -2%;
  left: 29%;
}

.star{
  position: absolute;
  top: 43%;
  left: 20%;
}

.star2{
  position: absolute;
  top: 64%;
  left: 65%;
  transform: rotate(80deg);
}

.you{
  position: absolute;
  top: 43%;
  left: 25%;
}

.lose{
  position: absolute;
  top: 55%;
  left: 57%;
}
