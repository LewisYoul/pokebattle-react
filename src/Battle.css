@import url('https://fonts.googleapis.com/css?family=Bangers|Russo+One');

*{
  margin: 0 auto;
  padding: 0;
}

body {
  box-sizing: border-box;
  padding-bottom: 2cm;
}

.p1name{
  font-family: Bangers;
  font-size: 40px;
  position: absolute;
  top:12%;
  left:8%;
  color: #17181a;
}

.p2name{
  font-family: Bangers;
  font-size: 40px;
  position: absolute;
  top:12%;
  left:77.5%;
  color: #17181a;
}

.circle{
  background-color: white;
  width: 200px;
  height: 100px;
  border-radius: 50%;
  position: absolute;
  top: 7%;
  left: 5%;
  box-shadow: 6px 8px #222529;
}

.circle2{
  background-color: white;
  width: 200px;
  height: 100px;
  border-radius: 50%;
  position: absolute;
  top: 7%;
  left: 75%;
  box-shadow: 6px 8px #222529;
}

.power{
  position: absolute;
  left:38%;
  top:12%;
}

.p1{
  position: absolute;
  top: 25%;
  left: 20%;
  -webkit-clip-path: circle(50% at 50% 50%);
  clip-path: circle(50% at 50% 50%);
}

.p2{
  position: absolute;
  top: 25%;
  left: 66%;
  -webkit-clip-path: circle(50% at 50% 50%);
  clip-path: circle(50% at 50% 50%);
}

.h1{
  font-family: Bangers;
  font-size: 35px;
  position: absolute;
  left: 20%;
  top: 84%;
}

.h2{
  font-family: Bangers;
  font-size: 35px;
  position: absolute;
  left: 63%;
  top: 84%;
}

.attack1{
  font-family: Bangers;
  font-size: 35px;
  text-align: center;
  padding: 10px 10px;
  background-color: #ffbf00;
  border: none;
  width: 170px;
  height: 85px;
  position: absolute;
  left: 12%;
  top: 62%;
  box-shadow: 3px 8px #1f2226;
  outline: none;
}


.attack1:active{
  transform: translateY(0.3rem);
  box-shadow: 0.2rem 0.3rem #1f2226;
}

.attack2{
  font-family: Bangers;
  font-size: 30px;
  -webkit-text-stroke-width: 1.7px;
 -webkit-text-stroke-color: #c62f2f;
  color: #fdea34;
  background-color: #19717e;
  border: none;
  text-align: center;
  padding: 10px 10px;
  width: 170px;
  height: 85px;
  position: absolute;
  left: 28%;
  top: 62.1%;
  box-shadow: 4px 8px #1f2226;
  outline: none;
}

.attack2:active{
  transform: translateY(0.3rem);
  box-shadow: 0.2rem 0.2rem #1f2226;
}

.attack3{
  font-family: Bangers;
  font-size: 40px;
  -webkit-text-stroke-width: 1.7px;
 -webkit-text-stroke-color: #c62f2f;
  color: #fdea34;
  background-color: #19717e;
  border: none;
  text-align: center;
  padding: 10px 10px;
  width: 170px;
  height: 85px;
  position: absolute;
  left: 56%;
  top: 62%;
  box-shadow: -4px 9px #1f2226;
  outline: none;
}

.attack3:active{
  transform: translateY(0.3rem);
  box-shadow: -0.2rem 0.2rem #1f2226;
}

.attack4{
  font-family: Bangers;
  font-size: 30px;
  text-align: center;
  background-color: #ffbf00;
  border: none;
  padding: 10px 10px;
  width: 170px;
  height: 85px;
  position: absolute;
  left: 72%;
  top: 62%;
  box-shadow: -4px 8px #1f2226;
  outline: none;
}

.attack4:active{
  transform: translateY(0.3rem);
  box-shadow: -0.2rem 0.2rem #1f2226;
}
