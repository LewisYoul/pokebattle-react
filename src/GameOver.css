@import url('https://fonts.googleapis.com/css?family=Bangers|Russo+One');

*{
  margin: 0 auto;
  padding: 0;
  box-sizing: inherit;
}

.moveimage{
  position: absolute;
  top: 35%;
  left: 39%;
}



.button{
  font-family: Bangers;
  font-size: 35px;
  color: yellow;
  position: absolute;
  left: 39%;
  top: 76%;
  width: 200px;
  padding: 10px 20px;
  outline: none;
  border: 2px solid #fdea34;
  background-color: #19717e;
  box-shadow: 6px 8px #1f2226;
}
