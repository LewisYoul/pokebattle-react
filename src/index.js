import React from 'react';
import ReactDOM from 'react-dom';
import './index.css';
import Player from './App';
import Api from './Api';
import registerServiceWorker from './registerServiceWorker';

ReactDOM.render(<h1><App></App></h1>, document.getElementById('root'));

registerServiceWorker();
// ReactDOM.render(<Api />, document.getElementById('root'));
// registerServiceWorker();
// ReactDOM.render(<Player name1='Marco' hp1={100} action1="fuck off" name2='Cristhian' hp2={100} action2="sing"/>, document.getElementById('root'));
