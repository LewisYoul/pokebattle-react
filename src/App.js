import React, { Component } from 'react';
// import logo from './logo.svg';
import './App.css';

class Player extends Component {
  constructor(props){
    super(props);
    this.state = {
      hp1: this.props.hp1,
      hp2: this.props.hp2
    };
    this.action1 = this.action1.bind(this);
    this.action2 = this.action2.bind(this);
  }

  action1() {
     this.setState({hp2: this.state.hp2 - 10});
  }

  action2() {
     this.setState({hp1: this.state.hp1 - 10});
  }

  render() {
    return (
      <span>
        <div id='player1'>
          <p>{this.props.name1}</p>
          <p>Hit Points: {this.state.hp1}</p>
          <button onClick={this.action1}>{this.props.action1}</button>
        </div>
        <div id='player2'>
          <p>{this.props.name2}</p>
          <p>Hit Points: {this.state.hp2}</p>
          <button onClick={this.action2}>{this.props.action2}</button>
        </div>
      </span>
    );
  }
}

export default Player;
