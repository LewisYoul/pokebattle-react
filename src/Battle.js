import './App.css';
import React, { Component } from 'react';
import { GameOver } from './GameOver'
import ToggleDisplay from 'react-toggle-display';

// import { Api } from './Api'
// import Player from './Player/Player'

class Battle extends Component {
  constructor(props){
    super(props);
    this.state = {
      hp1: this.props.player1.hp,
      hp2: this.props.player2.hp,
      strength1: this.props.player1.strength,
      strength2: this.props.player2.strength,
      showGameOver: false,
      show: true,
      image1: this.props.player1.image.url,
      loser: "here"
    };
    this.player1PrimaryAttack = this.player1PrimaryAttack.bind(this);
    this.player2PrimaryAttack = this.player2PrimaryAttack.bind(this);
    this.player1SecondaryAttack = this.player1SecondaryAttack.bind(this);
    this.player2SecondaryAttack = this.player2SecondaryAttack.bind(this);
  }


  randomAttack1 (strength) {
    return Math.floor(((Math.random() * 4) + 6)*(strength/100))
  }

  randomAttack2 (strength) {
    return Math.floor(((Math.random() * 4) + 9)*(strength/100))
  }


  player1PrimaryAttack() {
    const stop = (this.state.hp2 >= 0)
    stop ? this.setState({hp2: this.state.hp2 - this.randomAttack1(this.state.strength1)}) : this.setState({showGameOver: true, show: false})
    this.setState({loser: "http://localhost:4000" + this.state.image2});
  }

  player1SecondaryAttack() {
    const stop = (this.state.hp2 >= 0)
    stop ? this.setState({hp2: this.state.hp2 - this.randomAttack2(this.state.strength1)}) : this.setState({showGameOver: true, show: false})
    this.setState({loser: "http://localhost:4000" + this.state.image2});
  }

  player2PrimaryAttack() {
    const stop = (this.state.hp1 >= 0)
    stop ? this.setState({hp1: this.state.hp1 - this.randomAttack1(this.state.strength2)}) : this.setState({showGameOver: true, show: false})
    this.setState({loser: "http://localhost:4000" + this.state.image1});

  }
  player2SecondaryAttack() {
    const stop = (this.state.hp1 >= 0)
    stop ? this.setState({hp1: this.state.hp1 - this.randomAttack2(this.state.strength2)}) : this.setState({showGameOver: true, show: false})
    this.setState({loser: "http://localhost:4000" + this.state.image1});
  }

  render() {
    return (
      <span>
        <ToggleDisplay show={this.state.show}>
          <div id='player1'>
            <p>{this.props.player1.name}</p>
            <p>Hit Points: {this.state.hp1}</p>
            <img src={"http://localhost:4000" + this.props.player1.image.url} height="100px" width="100px" />
            <button onClick={this.player1PrimaryAttack}>{this.props.player1.actions[0].name}</button>
            <button onClick={this.player1SecondaryAttack}>{this.props.player1.actions[1].name}</button>
          </div>
          <br />
          <br />
          <div id='player2'>
            <p>{this.props.player2.name}</p>
            <p>Hit Points: {this.state.hp2}</p>
            <img src={"http://localhost:4000" + this.props.player2.image.url} height="100px" width="100px"/>
            <button onClick={this.player2PrimaryAttack}>{this.props.player2.actions[0].name}</button>
            <button onClick={this.player2SecondaryAttack}>{this.props.player2.actions[1].name}</button>
          </div>
        </ToggleDisplay>
        <ToggleDisplay show={this.state.showGameOver}>
          <GameOver image={this.state.loser} />
        </ToggleDisplay>
      </span>

    );
}
}
export default Battle;
