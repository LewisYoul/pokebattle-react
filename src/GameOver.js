import React, { Component } from 'react';
import ToggleDisplay from 'react-toggle-display';
import App from './App';
import './GameOver.css'

export class GameOver extends Component {
  constructor(props){
    super(props);
    this.state = {
      startNew: false,
      gameOver: true
    }
    this.playAgain = this.playAgain.bind(this)
  }

  playAgain(){
    this.setState({
      startNew: true,
      gameOver: false
    });
  }

  render(){
    return(
      <div>
        <ToggleDisplay show={this.state.gameOver}>
          <img class="gameover" src='images/gameover.jpeg' height="250px" width="400px"/>
          <div className="moveimage">
          <img className="image" src={this.props.image} height="200px" width="200px" />
          </div>
          <img class="you" src='images/you-.png' height="200px" width="200px"/>
          <img class="lose" src='images/lose.png' height="50px" width="100px"/>
          <button className="button" onClick={this.playAgain}>Play Again?</button>
        </ToggleDisplay>
        <ToggleDisplay show={this.state.startNew}>
          <App />
        </ToggleDisplay>
      </div>

    )
  }
}
