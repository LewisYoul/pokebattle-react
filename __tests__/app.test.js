import React from 'react'
import ReactDOM from 'react-dom'
import App from '../src/App';
// import {concatStrings} from '../src/App.js';
import ReactTestUtils from 'react-dom/test-utils';
import '../src/setupTests.js'
import { concatStrings } from '../src/App'
import renderer from 'react-test-renderer';

describe('Addition', () => {
  it('knows that 2 and 2 make 4', () => {
    expect(2 + 2).toBe(4);
  });
});

describe('FEATURE - App', () => {
  it('it renders Loading Players... to the page', () => {
    const rendered = renderer.create(
      <App />
    );
    console.log(rendered.toJSON())
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
