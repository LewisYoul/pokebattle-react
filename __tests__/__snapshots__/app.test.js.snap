// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`String Concatenation concatenates two strings 1`] = `
<p>
  Loading Players...
</p>
`;
